ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleDBTest.java

```diff
@@ -0,0 +1,145 @@
+import android.arch.persistence.room.Room;
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.cru.godtools.articles.aem.model.Article;
+import org.cru.godtools.articles.aem.model.Attachment;
+import org.cru.godtools.articles.aem.model.ManifestAssociation;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import timber.log.Timber;
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ArticleDBTest implements LifecycleOwner {
+
+    private ArticleDao mArticleDao;
+    private AttachmentDao mAttachmentDao;
+    private ManifestAssociationDao mAssociationDao;
+    private ArticleRoomDatabase db;
+
+    private Observer<List<Article>> articleObserver;
+
+    @Before
+    public void createDb() {
+        Context context = InstrumentationRegistry.getTargetContext();
+        db = Room.inMemoryDatabaseBuilder(context,
+                ArticleRoomDatabase.class).allowMainThreadQueries().build();
+        mArticleDao = db.articleDao();
+        mAttachmentDao = db.attachmentDao();
+        mAssociationDao = db.manifestAssociationDao();
+
+    }
+
+    @After
+    public void closeDb() {
+        db.close();
+    }
+
+    @Test
+    public void useAppContext() {
+        // Context of the app under test.
+        Context appContext = InstrumentationRegistry.getTargetContext();
+
+        List<Article> articles = new ArrayList<>();
+
+        for (int i = 0; i < 12; i++) {
+            Article article = new Article();
+
+            article.mContent = "<p> The Body </>";
+            try {
+                article.mDateCreated = new SimpleDateFormat("E MMM dd yyyy HH:mm:ss zz")
+                        .parse("Fri Jun 08 2018 18:55:00 GMT+0000").getTime();
+                article.mDateUpdated = new SimpleDateFormat("E MMM dd yyyy HH:mm:ss zz")
+                        .parse("Sat May 19 2018 00:23:39 GMT+0000").getTime();
+            } catch (ParseException e) {
+                Timber.e(e);
+                article.mDateCreated = Calendar.getInstance().getTimeInMillis();
+            }
+            Calendar calendar = Calendar.getInstance();
+            calendar.getTime();
+            calendar.setTimeInMillis(article.mDateCreated);
+            Date date = new Date(article.mDateCreated);
+            date.getYear();
+            date.getDate();
+            date.getHours();
+            date.getTimezoneOffset();
+            article.mTitle = " The title = " + i;
+            mArticleDao.insertArticle(article);
```

Comment:
you are inserting new articles, but never verifying that they are actually being stored in the database. This test would have limited utility